inc/azure_c_shared_utility/refcount.h

```diff
@@ -69,6 +69,9 @@ static type* REFCOUNT_TYPE_DECLARE_CREATE(type) (void)
 #ifndef DEC_REF
 #error refcount_os.h does not define DEC_REF
 #endif // !DEC_REF
+#ifndef INIT_REF
+#error refcount_os.h does not define INIT_REF
+#endif // !INIT_REF
```

Comment:
Why are these necessary?  Won't the linker complain if the defines are not there?

Reply:
I think Roy added these to have some ... nicer error messages. I do not remember them existing in the first version of refcount.h done eons ago. But anyhow, I guess I am Ok with having them for slightly better errors. Was not that painful to add :-)

---
In reply to: [201130723](https://github.com/Azure/azure-c-shared-utility/pull/165#discussion_r201130723) [](ancestors = 201130723)